docs/src/python/data_types.rst

```diff
@@ -50,3 +50,18 @@ The default floating point type is ``float32`` and the default integer type is
    * - ``float32``
      - 4 
      - 32-bit float
+
+Categories
+----------
+
+.. autosummary::
+  :toctree: _autosummary
+
+  complexfloating
+  floating
+  generic
+  inexact
+  integer
+  number
+  signedinteger
+  unsignedinteger
```

Comment:
The doc here is not that great, it just has the same line next to each:

<img width="813" alt="Screenshot 2024-03-01 at 2 30 37 PM" src="https://github.com/ml-explore/mlx/assets/1542805/8eb2b584-010d-48c2-8f2d-e145c23f5561">


Maybe we could do it as a table with a line below `Categories` explaining that these are the :obj:`DtypeCategory`? (Kind of like how we did it with the dtypes above).

Reply:
Good point. I'd suggest to simply embed the docstring of `DtypeCategory`, which documents the hierarchy already  to avoid duplicating the information. What do you think: 

![image](https://github.com/ml-explore/mlx/assets/1847260/870ce2e7-3081-4d62-a4b6-ee9bc4704d5e)

Reply:
This sounds good to me!